src/Raven.Server/Documents/Handlers/StreamingHandler.cs

```diff
@@ -70,12 +71,30 @@ public Task SteamQueryHead()
             using (var token = CreateTimeLimitedQueryToken())
             using (Database.DocumentsStorage.ContextPool.AllocateOperationContext(out DocumentsOperationContext context))
             {
-                var query = IndexQueryServerSide.Create(HttpContext, GetStart(), GetPageSize(), context);
+                var documentId = GetStringQueryString("fromDocument", false);
+                string overrideQuery = null;
+                if (string.IsNullOrEmpty(documentId) == false)
+                {
+                    Document document;
+                    using (context.OpenReadTransaction())
+                    {
+                        document = Database.DocumentsStorage.Get(context, documentId);                    
+                        if (document == null)
+                        {
+                            throw new DocumentDoesNotExistException($"Was request to stream a query taken from {documentId} document, but it does not exist.");
+                        }
+                        if (document.Data.TryGetMember("Query", out var obj))
```

Comment:
Can use `out string str` here.
But ends up being the same thing.

Reply:
@ayende it is not a string always, i'm my tests it was LazyCompressedString or something like that.

Reply:
Oh, you are using `TryGetMember`. IIRC, `TryGet` or `TryGetValue` also handle automatic conversions.

Reply:
https://github.com/ravendb/ravendb/pull/4904